src/triage/component/postmodeling/contrast/group_evaluator.py

```diff
@@ -0,0 +1,325 @@
+        self.train_matrix_uuid = model_metadata['train_matrix_uuid']
+        self.test_matrix_uuid = model_metadata['test_matrix_uuid']
+        self.train_matrix = None
+        self.pred_matrix = None
+        self.preds = None
+        self.feature_importances = None
+        self.model_metrics = None
+
+    def __str__(self):
+        s = 'Model collection object for model_ids: {}'.format(self.model_id)
+        s += '\n Model Groups: {}'.format(self.model_group_id)
+        s += '\n Model types: {}'.format(self.model_type)
+        s += '\n Model hyperparameters: {}'.format(self.hyperparameters)
+        s += '\Matrix hashes (train,test): {}'.format((self.train_matrix_uuid,
+                                                      self.test_matrix_uuid))
+        return s
+
+    def _load_predictions(self):
+        preds = pd.read_sql("""
+                SELECT model_id,
+                       entity_id,
+                       as_of_date,
+                       score,
+                       label_value,
+                       COALESCE(rank_abs, RANK() 
+                       OVER(PARTITION BY model_id ORDER BY score DESC)) AS rank_abs,
+                       rank_pct,
+                       test_label_timespan
+                FROM test_results.predictions
+                WHERE model_id IN {}
+                AND label_value IS NOT NULL
+                """.format(tuple(self.model_id)),
+                con=db_engine)
+        self.preds = preds
+
+    def _load_feature_importances(self):
+        features = pd.read_sql("""
+               SELECT model_id,
+                      feature,
+                      feature_importance,
+                      rank_abs
+               FROM train_results.feature_importances 
+               WHERE model_id IN {}
+               """.format(tuple(self.model_id)),
+               con=db_engine)
+        self.feature_importances = features
+
+    def _load_metrics(self):
+        model_metrics = pd.read_sql("""
+                SELECT model_id,
+                       metric,
+                       parameter,
+                       value,
+                       num_labeled_examples,
+                       num_labeled_above_threshold
+                       num_positive_labels
+                FROM test_results.evaluations
+                WHERE model_id IN {}
+                """.format(tuple(self.model_id)),
+                con=db_engine)
+        self.model_metrics = model_metrics
+
+    def _rank_corr_df(self,
+                      model_pair,
+                      top_n,
+                      type):
+        '''
+        Calculates ranked correlations for ranked observations and features
+        using the stats.spearmanr scipy module.
+        Arguments: 
+            - model_pair (tuple): tuple with model_ids
+              observations or features
+            - top_n (int): number of rows to rank (top-k model)
+        '''
+
+        if type not in ['predictions', 'features']:
```

Comment:
Though *really* not a big deal, you might notice that `type` is being syntax-highlighted – because it's a Python built-in. Plenty of function signatures do precisely the same, but it *might* be nice to avoid that built-in, (though not reserved), word, for example by just making it more specific, say `rank_type`.